### PR TITLE
Added Quick Start Guide Documentation

### PR DESCRIPTION
**Pull request by csteavu**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-instrument-addon-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Added Quick Start Guide Documentation markdown file.
The file contains: 

- the information from the html quick start

- the Scripting API information

- examples for the Scripting API

### Why should this Pull Request be merged?

Html Quick Start Guide did not contain the Scripting API information;

### What testing has been done?

Visual Check of the MD file

### CODE REVIEW

**Comment by bogdanp-ni** [comment]
Please remove the old docs and update the links (from Main Page Help and README.md) to point to the new Markdown docs.

**Review by bogdanp-ni — APPROVED** [review]
I'm ok with this, as long as the links are fixed (see prev. comment).

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+## With this API, users can:<br/>
+- create a new Instrument from a previously exported configuration:
+
+    Node used: **Create Instrument Addon From Configuration File**
+
+![Create Instrument Addon From Configuration File](./Images/image040.png)
+
+- create a new Instrument from scratch<br/>
+    Node used: **Create Instrument Addon**
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:193]: A new line needs to be added here for better rendering and consistency with the other examples.

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+Users can add flexibility and reusability to their systems by creating and configuring system definitions through the Scripting API.
+
+## With this API, users can:<br/>
+- create a new Instrument from a previously exported configuration:
+
+    Node used: **Create Instrument Addon From Configuration File**
+
+![Create Instrument Addon From Configuration File](./Images/image040.png)
+
+- create a new Instrument from scratch<br/>
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:192]: A new line needs to be added here for better rendering and consistency with the other examples.

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+![Add Response](./Images/image044.png)
+
+    Node used: **Create Message**<br/>
+
+![Create Message](./Images/image045.png)
+
+- Create Channels
+
+    Node used: **Create Global MetaData Channel**<br/>
+![Create Global MetaData Channel](./Images/image046.png)
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:215]: image46.png needs to be updated to show all the different types of Data Channels allowed. Currently it shows only one: Single Channel Enumerated List.

image

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+    Node used: **Set Protocol**   
+
+![Get and Set Protocol](./Images/image049.png)
+
+- Get and Set the PCL Decimation
+
+    Node used: **Get PCL Decimation**
+    Node used: **Set PCL Decimation**   
+
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:237]: This rendering does not look consistent.

image

We can change the description to something like: "Nodes used: **Get PCL Decimation** and **Set PCL Decimation**" and do the same thing the Get Protocol and Set Protocol.

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+![Get and Set PCL Decimation](./Images/image050.png)
+
+- Make use of the Protocols subpalette - containing the Get Protocol Type node and factory nodes for each supported protocol
+   
+A Protocol object is needed as input when creating a new Instrument Addon<br/>
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:242]: A new line is needed here for better rendering.

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+   - RS-xxx to ConvBox
+   - VISA
+   - TCP
+   - UDP
+   - AK RS-xxx
+   - AK TCP
+   
+    Factory Nodes used:
+
+![Factories](./Images/image052.png)
```
Comment (phutanu) [inline Quick Start Documentation/Instrument Addon Quick Start Guide.md:254]: For whatever reason the above line is rendered different than the other lines. This breaks the consistency of the file.

image

**Comment by phutanu** [comment]
[Recommendation] what do you think if we replace the "Settings" typedef control input with a "Settings" constant?
This will be helpful for the users to see what settings are available for each node type and polymorphic instance.

**Review by phutanu — CHANGES_REQUESTED** [review]
Please check my above comments.

**Comment by csteavu** [comment]
> [Recommendation] what do you think if we replace the "Settings" typedef control input for all the Scripting single nodes from the section "With this API, users can:" with a "Settings" constant?
> I put this as a recommendation because I know it will take a lot of work, as we need to update a lot of png files.
> This change will be helpful for the users to see what settings are available for each node type and polymorphic instance.
> 
> Like the examples from the Scripting API HLD Doc:
> image

This recommendation brings a lot of value. As discussed offline, this will be implemented along with the rework that will arise from the API Review

**Comment by bogdanp-ni** [comment]
> > [Recommendation] what do you think if we replace the "Settings" typedef control input for all the Scripting single nodes from the section "With this API, users can:" with a "Settings" constant?
> > I put this as a recommendation because I know it will take a lot of work, as we need to update a lot of png files.
> > This change will be helpful for the users to see what settings are available for each node type and polymorphic instance.
> > Like the examples from the Scripting API HLD Doc:
> > image
> 
> This recommendation brings a lot of value. As discussed offline, this will be implemented along with the rework that will arise from the API Review

+1 for this getting added with the API review fixes, but please add another task specifically for this request. Otherwise, it will be lost in the multitude of small updates we'll have to do.

File: Quick Start Documentation/Instrument Addon Quick Start Guide.md (hunk 2 above)
Comment (phutanu) [reply Quick Start Documentation/Instrument Addon Quick Start Guide.md]: I'm ok with the changes, but just a new line here would make a huge visual difference.

image

**Review by phutanu — APPROVED** [review]
> I'm ok with the changes, but just a new line where the red arrow is would make a huge visual difference.
> 
> image

**Pull request merged**